test/ex_doc/cli_test.exs

```diff
@@ -35,6 +35,41 @@ defmodule ExDoc.CLITest do
     assert io == "ExDoc v#{ExDoc.version()}\n"
   end
 
+  describe "--warnings-as-errors" do
+    test "exits with code 0 when no warnings" do
+      {[html, epub], _io} = run(["ExDoc", "1.2.3", @ebin, "--warnings-as-errors"])
+
+      assert html ==
+               {"ExDoc", "1.2.3",
+                [formatter: "html", apps: [:ex_doc], source_beam: @ebin, warnings_as_errors: true]}
+
+      assert epub ==
+               {"ExDoc", "1.2.3",
+                [formatter: "epub", apps: [:ex_doc], source_beam: @ebin, warnings_as_errors: true]}
+    end
+
+    test "exits with 1 with warnings" do
+      ExDoc.WarningCounter.increment()
+
+      fun = fn ->
+        io =
+          capture_io(:stderr, fn ->
+            {[{"ExDoc", "1.2.3", opts}, _], _io} =
+              run(["ExDoc", "1.2.3", @ebin, "--warnings-as-errors"])
+
+            assert Keyword.get(opts, :warnings_as_errors)
```

Comment:
I'm pretty sure that this test never actually hits this `assert` because of the `exit`. If I throw a `IO.puts` right above it, I never actually see it.

Reply:
Yeah your right 😞  I wonder how I can capture_io and the exit in the same go?

Reply:
I'm not sure if there's a way to capture the output of a function that `exit`s. I believe this would be hitting [this guy](https://github.com/elixir-lang/elixir/blob/1e4ed32eda7957d469860e0143b5e5021538b477/lib/ex_unit/lib/ex_unit/capture_io.ex#L270-L282) which throws away the IO if the function throws.

Reply:
This works:

```elixir
      fun = fn ->
        run(["ExDoc", "1.2.3", @ebin, "--warnings-as-errors"])
      end

      io =
        capture_io(:stderr, fn ->
          assert catch_exit(fun.()) == {:shutdown, 1}
        end)

      assert io =~
               "Doc generation failed due to warnings while using the --warnings-as-errors option\n"
```

Reply:
@eksperimental Thanks! just pushed that up

Reply:
Awesome! Let me/us know if you need help with anything.